covidWallet/app/components/SettingsScreen.js

```diff
@@ -1,13 +1,145 @@
+    },
+    key: '3'
+  }
+}
+
+export default function SettingsScreen () {
+  const [settingsData, setSettingsData] = useState(settingLocalData)
 
-function SettingsScreen() {
+  const toggleSwitch = (parent, child) => {
+    const tempSettings = { ...settingsData }
+    tempSettings[parent][child].value = !tempSettings[parent][child].value
+    setSettingsData({ ...tempSettings })
+  }
   return (
-    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
-      <Text>SettingsScreen</Text>
+    <View style={styles.container}>
+      <FlatList
+        data={Object.keys(settingsData)}
+        keyExtractor={(item, index) => settingsData[item].key}
+        renderItem={({ item }) => {
+          const parent = item
+          const parentData = settingsData[parent]
+          return (
+            <View>
+              <Text style={styles.parentItem}>{parent}</Text>
+              <FlatList
+                data={Object.keys(parentData)}
+                keyExtractor={(item, index) => index.toString()}
+                renderItem={({ item }) => {
+                  const childData = settingsData[parent][item]
+                  if (item !== 'key' && item !== '') {
+                    if (childData.value !== 'None') {
+                      if (childData.type === 'Text') {
+                        return (
+                          <TextTypeView startValue={item + ':  ' + childData.value} endValue='Edit' endIcon='' />
+                        )
+                      } else if (childData.type === 'Radio') {
+                        return (
+                          <TextTypeView startValue={item + ':  ' + childData.value} endIcon='right' />
+                        )
+                      } else if (childData.type === 'Boolean') {
+                        return (
+                          <BooleanTypeView
+                            parentValue={parent}
+                            startValue={item}
+                            endValue={childData.value}
+                            toChangeValue={childData.value}
+                            valueHandler={toggleSwitch}
+                          />
+                        )
+                      } else {
+                        return (
+                          <TextTypeView startValue={item + ':  ' + childData.value} endIcon='right' />
+                        )
+                      }
+                    } else {
+                      return (
+                        <TextTypeView startValue={item} endValue='Edit' endIcon='right' />
+
+                      )
+                    }
+                  }
+                }}
+              />
+            </View>
+          )
+        }}
+      />
+
     </View>
-  );
+  )
 }
 
-export default SettingsScreen;
\ No newline at end of file
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    paddingTop: 20,
+    paddingLeft: 5,
+    paddingRight: 5,
+    backgroundColor: '#f7f7f7'
```

Comment:
Colors in our app come from a central file. Please look at other components for reference